app/helpers/filter_categories_helper.rb

```diff
@@ -40,7 +40,7 @@ def subcategories_by_category(category_id)
     fetch_categories if @categories.nil?
 
     @categories.select { |category|
-      category.depth == 1 and category.parent_id == category.to_i
+      category.depth == 1 and category.parent_id == category.id.to_i
```

Comment:
```suggestion
      category.depth == 1 and category.parent_id == category_id.to_i
```

I think this is what we want, I'm guessing I messed this up when I changed this file a bit